docs/languages.md

```diff
@@ -36,6 +36,33 @@ _Troubleshooting: Variants missing from this file cause onboarding to fail to id
 
 Note: Historically this mapping file was used to determine the user's language variant preference in Chinese and Serbian based on the user's OS language settings. The new language variant feature allows the user to choose variants explicitly.
 
+#### Locale to language variant mapping details
+A locale identifier in the OS can include:
+
+- A language identifier on its own. For example: "zh" for _Chinese_.
+
+- Language and script identifiers. For example: "sr_Cyrl" for _Serbian, Cryillic_; "sr_Latn" for _Serbian, Latin_
+
+- Language, script, and region identifiers. For example: "zh_Hant_TW" for _Taiwanese, Traditional_; "zh_Hans_HK" for _Hong Kong, Simplified_
+
+In Chinese, the region affects the Wikipedia langauge variant. So, the mapping file specifies regions in the most deeply nested dictionary.
+
+In all other languages with variants so far, the script identifier is enough to identify the variant, and does not change regardless of region.
+For these languages, only the "default" key is present in the region dictionary.
+
+Also note that not all scripts/variants supported by a particular Wikipedia language are available identifiers in the OS.
+In these cases, finding the language in the user's OS preferences implies a particular language variant.
+
+Note that "gan" and "crh" are not available in the OS, although their scripts are. A default language variant value is provided in case a future OS includes these locale identifiers.
```

Comment:
I may be getting things mixed up but I also couldn't find any Kurdish variants as an OS language to choose (ku-latn and ku-arab) either. All I see is Kurdish, Sorani which mapped to the `ckb` language code. Just noting in case it's worth mentioning here.

Reply:
I found the same thing. I had researched by filtering the list of `Locale.availableIdentifiers` which does include Kurdish identifiers. I updated the note to add that Kurdish is currently not selectable in the OS language settings.

That is different from 'gan' and 'crh' which do not appear in the Locale available identifiers list.